### ISSUE TITLE
Events need to extend AbstractEvent

### ISSUE THREAD

**Issue by RobertHerhold** [Erigitic/TotalEconomy#96]
With this event listener:
```
@Listener
public void event(Event event) {

}
```

I get this exception:

```
[23:08:46 ERROR] [Sponge]: Could not pass TEEconomyTransactionEvent to Plugin{id=t, name=Test, version=1.0.0, source=mods/test-1.0.0.jar}
java.lang.ClassCastException: com.erigitic.config.TEEconomyTransactionEvent cannot be cast to org.spongepowered.api.event.impl.AbstractEvent
	at org.spongepowered.common.event.SpongeEventManager.post(SpongeEventManager.java:304) [SpongeEventManager.class:1.10.2-5.0.0-BETA-129]
	at org.spongepowered.common.event.SpongeEventManager.post(SpongeEventManager.java:320) [SpongeEventManager.class:1.10.2-5.0.0-BETA-129]
	at com.erigitic.config.TEAccount.deposit(TEAccount.java:133) [TEAccount.class:?]
	at org.spongepowered.api.service.economy.account.Account.deposit(Account.java:294) [Account.class:1.10.2-5.0.0-BETA-129]
	at com.erigitic.commands.AdminPayCommand.execute(AdminPayCommand.java:51) [AdminPayCommand.class:?]
	at org.spongepowered.api.command.spec.CommandSpec.process(CommandSpec.java:351) [CommandSpec.class:1.10.2-5.0.0-BETA-129]
	at org.spongepowered.api.command.dispatcher.SimpleDispatcher.process(SimpleDispatcher.java:333) [SimpleDispatcher.class:1.10.2-5.0.0-BETA-129]
	at org.spongepowered.common.command.SpongeCommandManager.process(SpongeCommandManager.java:291) [SpongeCommandManager.class:1.10.2-5.0.0-BETA-129]
	at net.minecraft.command.ServerCommandManager.func_71556_a(SourceFile:82) [bd.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147361_d(SourceFile:825) [me.class:?]
	at net.minecraft.network.NetHandlerPlayServer.func_147354_a(SourceFile:812) [me.class:?]
	at net.minecraft.network.play.client.CPacketChatMessage.func_148833_a(SourceFile:37) [im.class:?]
	at net.minecraft.network.play.client.CPacketChatMessage.func_148833_a(SourceFile:9) [im.class:?]
	at org.spongepowered.common.network.PacketUtil.onProcessPacket(PacketUtil.java:122) [PacketUtil.class:1.10.2-5.0.0-BETA-129]
	at net.minecraft.network.PacketThreadUtil$1.redirect$onProcessPacket$0(SourceFile:39) [fl$1.class:?]
	at net.minecraft.network.PacketThreadUtil$1.run(SourceFile:13) [fl$1.class:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_102-ea]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_102-ea]
	at net.minecraft.util.Util.func_181617_a(SourceFile:45) [h.class:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(SourceFile:140) [MinecraftServer.class:?]
	at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(SourceFile:338) [ld.class:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(SourceFile:554) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer.run(SourceFile:458) [MinecraftServer.class:?]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_102-ea]

```

**Comment by Erigitic** [comment]
I believe this is caused by you using `Event` instead of a more descriptive event such as `ChangeBlockEvent.Break`.

**Comment by RobertHerhold** [comment]
@Erigitic I reported this same issue to Flibio's EconomyLite: Flibio/EconomyLite#18. Extending `AbstractEvent` resolved the issue for me, which seems to be caused by [this commit in SpongeCommon](https://github.com/SpongePowered/SpongeCommon/commit/4d8439796cdd4b76569a97286ef2e04cf07976c3#diff-0dd212c5ab4f79ff70118d9539bc8453R304).

**Comment by Erigitic** [comment]
Alright, I will take a look into this during the week. Thanks for the report.

**Comment by Erigitic** [comment]
The change has been made and will be released with version 1.5.1 of Total Economy. Thanks again for the report.
